bin/bridge-client/src/modes/operator/task_manager.rs

```diff
@@ -18,57 +18,56 @@ use strata_storage::ops::{bridge_duty::BridgeDutyOps, bridge_duty_index::BridgeD
 use tokio::{task::JoinSet, time::sleep};
 use tracing::{error, info, trace, warn};
 
-pub(super) struct TaskManager<L2Client, TxBuildContext, Bcast>
+pub(super) struct TaskManager<TxBuildContext, Bcast>
 where
-    L2Client: StrataApiClient + Sync + Send,
     TxBuildContext: BuildContext + Sync + Send,
     Bcast: Broadcaster,
 {
-    pub(super) exec_handler: Arc<ExecHandler<L2Client, TxBuildContext>>,
+    pub(super) exec_handler: Arc<ExecHandler<TxBuildContext>>,
     pub(super) broadcaster: Arc<Bcast>,
     pub(super) bridge_duty_db_ops: Arc<BridgeDutyOps>,
     pub(super) bridge_duty_idx_db_ops: Arc<BridgeDutyIndexOps>,
 }
 
-impl<L2Client, TxBuildContext, Bcast> TaskManager<L2Client, TxBuildContext, Bcast>
+impl<TxBuildContext, Bcast> TaskManager<TxBuildContext, Bcast>
 where
-    L2Client: StrataApiClient + Sync + Send + 'static,
     TxBuildContext: BuildContext + Sync + Send + 'static,
     Bcast: Broadcaster + Sync + Send + 'static,
 {
     pub(super) async fn start(&self, duty_polling_interval: Duration) -> anyhow::Result<()> {
         loop {
-            let RpcBridgeDuties {
+            if let Ok(RpcBridgeDuties {
                 duties,
                 start_index,
                 stop_index,
-            } = self.poll_duties().await?;
-
-            let mut handles = JoinSet::new();
-            for duty in duties {
-                let exec_handler = self.exec_handler.clone();
-                let bridge_duty_ops = self.bridge_duty_db_ops.clone();
-                let broadcaster = self.broadcaster.clone();
-                handles.spawn(async move {
-                    process_duty(exec_handler, bridge_duty_ops, broadcaster, &duty).await
-                });
-            }
+            }) = self.poll_duties().await
+            {
```

Comment:
Changing this to an `if let` like this is a minor change but it makes the structure much different.  This should reason about the error conditions more explicitly.

I don't even see the error case being handled here, we need to do that.  From what I can tell, if this errors immediately (like because the connection is broken) then we just spin in a hot loop eating a CPU core until we don't encounter the error anymore.